web_export_view/controllers/controllers.py

```diff
@@ -40,7 +40,19 @@ def export_xls_view(self, data, token):
         data = json.loads(data)
         model = data.get('model', [])
         columns_headers = data.get('headers', [])
-        rows = data.get('rows', [])
+        rows = []
+        for row in data.get('rows', []):
+            row = [
+                ', '.join(
+                    map(
+                        lambda c: c.name_get()[0][1],
+                        request.env[cell['relation']].browse(cell['ids'])
+                    )
+                )
+                if isinstance(cell, dict) else cell
```

Comment:
don't guess stuff. Extend the protocol that it also passes the field names being exported. From this you can decide where to do anything at all and that relation to use. And don't copy around data, the id lists are references, you you can just replace those.